Content.Shared/Nyanotrasen/Shipyard/SharedShipyardSystem.cs

```diff
@@ -0,0 +1,45 @@
+using Content.Shared.Shipyard.Components;
+using Content.Shared.Containers.ItemSlots;
+using JetBrains.Annotations;
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Shipyard
+{
+    [NetSerializable, Serializable]
+    public enum ShipyardConsoleUiKey : byte
+    {
+        //Not currently implemented. Could be used in the future to give other factions a variety of shuttle options.
+        Shipyard,
+        Syndicate
```

Comment:
Not that it matters much now, but wouldn't you pass different state to indicate different availability options for shuttles rather than using an entirely different UiKey?

Reply:
For some reason UIkey needs to be implemented even if you only ever plan on having one state.

BUIs still confuse me a bit, but im fairly sure this is the correct method for having different UIs attached to the same prototype

Reply:
Right, the single UiKey's not an issue; that's just how it works from what I've learned. I'm pointing out here the `Syndicate` line and the comment about future shuttle options, and that it seems like you would set different UI state on Shipyard instead of making an entirely different UiKey for different shuttle options. Some sort of available shuttles list that's passed to the client.

Reply:
Hm I was under the impression that you could use the key to determine what ui to show.  So a syndicate console wouldn't just have a different list, it would have other options, for instance you wouldn't use bank account you would want to handle it differently.  I could just remove it for now

Reply:
You could do it that way if there were significant differences in how the UIs functioned, but if it's just a simple difference of what ships are available, passing in a different list as UiState probably makes more sense.

Reply:
Agreed.  I use entirely different UI functionality for the different ones.  For instance, if you wanted to let nukies pick their shuttle, you could have a UI that only allows a single "purchase" out of many options, so the UI would function completely different.  I'll remove the syndicate line for now.

Reply:
Nukie shuttle options sounds like a cool future use of this system. I.e. buy one big shuttle or buy a couple smaller shuttles.

Reply:
exactly. In my mind, nukies have war mechs that need special transport, so if you want to do warops(Mechwarrior style) you need to call in a mech drop ship!  While it will use the same systems and base components, the nukies' UI will look different because they don't need to worry about silly things like bank accounts, so they would probably get more like an option list style UI.

Or if you maybe wanted to give the Valets access to an uber terminal to let them call/send away taxis or something. Would use the same systems but a different looking UI.